tests/std/tests/GH_002045_put_time_changes_errno/test.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cerrno>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+using namespace chrono;
+
+int main() {
+    time_t t(system_clock::to_time_t(system_clock::now()));
+    {
+        ostringstream os;
+
+        assert(errno == 0);
+
+        std::tm tm;
+        {
+            const errno_t err = localtime_s(&tm, &t);
+            assert(err == 0);
+        }

```

Comment:
Suggestion: This test calls `localtime_s()` twice (for narrow and wide) with the same input. As this isn't the focus of the test (we're interested in `put_time()`'s behavior), it would be simpler to extract this to a helper function that returns a `tm`.